saleor/payment/utils.py

```diff
@@ -693,11 +693,37 @@ def payment_owned_by_user(payment_pk: int, user) -> bool:
     )
 
 
+def get_correct_event_types_based_on_request_type(request_type: str) -> List[str]:
+    if request_type == TransactionEventType.AUTHORIZATION_REQUEST:
+        return [
+            TransactionEventType.AUTHORIZATION_FAILURE,
+            TransactionEventType.AUTHORIZATION_ADJUSTMENT,
+            TransactionEventType.AUTHORIZATION_SUCCESS,
+        ]
+    elif request_type == TransactionEventType.CHARGE_REQUEST:
+        return [
+            TransactionEventType.CHARGE_FAILURE,
+            TransactionEventType.CHARGE_SUCCESS,
+        ]
+    elif request_type == TransactionEventType.REFUND_REQUEST:
+        return [
+            TransactionEventType.REFUND_FAILURE,
+            TransactionEventType.REFUND_SUCCESS,
+        ]
+    elif request_type == TransactionEventType.CANCEL_REQUEST:
+        return [
+            TransactionEventType.CANCEL_FAILURE,
+            TransactionEventType.CANCEL_SUCCESS,
+        ]
```

Comment:
The map with the request type to correct events types can be created and used here, instead of multiple `if, elif`